addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -2617,6 +2617,98 @@ var ImageSelection = AbstractField.extend({
         this._setValue($(event.currentTarget).data('key'));
     },
 });
+/**
+ * Alternative Widget to display a selection field with a preview image and link
+ * Example:
+ * <field widget="selection_with_preview"
+ * options="{'opt1': {'image_link': 'img1.png','preview_link': 'prv1.pdf'},
+ *           'opt2': {'image_link': 'img2.jpg','preview_link': 'prv2.html'}" />
+ */
+var SelectionWithPreview = AbstractField.extend({
+    supportedFieldTypes: ['selection'],
+    events: _.extend({}, AbstractField.prototype.events, {
```

Comment:
? put the image clickable ...  investigate if it is possible to reuse existing widget